### PR TITLE
Add tests for liveness fallbacks

### PR DESCRIPTION
**Pull request by elhesuu**
# Problem
We need to implement tests to assert liveness fallbacks work as expected:
```
  "desktop + no webcam -> x-device" scenario should be covered
  "desktop + safari (i.e. no `MediaRecorder`) -> selfie" scenario should be covered
```

# Solution

Since we can't enable the real webcam for tests, or emulate different browsers' implementation of `MediaRecorded`, the proposed approach is to stub `enumerateDevices` and unset `MediaRecorder` during if needed, the tests.

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [n/a] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [n/a] Have new automated tests been implemented?
- [n/a] Have new manual tests been written down?
- [x] Have tests passed locally?
- [n/a] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-449-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-449-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-449-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-449-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-449-pr-onfido-sdk-ui-onfido.surge.sh

File: test/features/step_definitions/sdk_steps.rb [hunk 1]
```diff
@@ -10,14 +10,23 @@
...
   steps %Q{
-    Given I navigate to the SDK using a webcam with "#{locale}"
+    Given I navigate to the SDK using liveness with "#{locale}"
     Then I click on primary_button (SDK)
   }
 end
 
+Given(/^I do( not)? have a camera$/) do |has_no_camera|
+  devices = !has_no_camera ? '{ kind: "video" }' : ''
+  @driver.execute_script('window.navigator.mediaDevices.enumerateDevices = () => Promise.resolve([' + devices + '])')
```
Comment (rfreitas) [inline test/features/step_definitions/sdk_steps.rb]: will this work on browsers without promises? If I remember correctly, for browsers without promise support we just assume there is no camera api available. However, that doesn't mean this test code won't break.

Comment (elhesuu) [reply test/features/step_definitions/sdk_steps.rb]: It won't but we are only using Chrome

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-449-pr-onfido-sdk-ui-onfido.surge.sh

File: test/features/step_definitions/sdk_steps.rb [hunk 2]
```diff
@@ -10,14 +10,23 @@
...
   i18n.load_locale(locale)
   steps %Q{
-    Given I navigate to the SDK using a webcam with "#{locale}"
+    Given I navigate to the SDK using liveness with "#{locale}"
     Then I click on primary_button (SDK)
   }
 end
 
+Given(/^I do( not)? have a camera$/) do |has_no_camera|
+  devices = !has_no_camera ? '{ kind: "video" }' : ''
```
Comment (stefaniacardenas) [inline test/features/step_definitions/sdk_steps.rb]: Can we avoid this double negative?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-449-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://449-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-449-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-449-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://449-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**